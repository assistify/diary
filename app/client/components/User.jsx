import React from 'react';
import { PropTypes } from 'prop-types';
import { TeamContext } from '../lib/teamContext';

export default function User(props) {
  const { username } = props;
  return (
    <TeamContext.Consumer>
      { teamContext => (
        <a
          className="c-user"
          href={`${teamContext.serverUrl}/direct/${username}`}
          rel="noopener noreferrer"
          target="_blank"
        >
          {`@${username}`}
        </a>
      )}
    </TeamContext.Consumer>
  );
}

User.propTypes = {
  username: PropTypes.string.isRequired
};

User.contextType = TeamContext;
