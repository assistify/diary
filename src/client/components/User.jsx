import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { TeamContext } from '../lib/teamContext';

export default class User extends Component {
  static propTypes = {
    username: PropTypes.string.isRequired
  }

  static contextType = TeamContext;

  constructor(props) {
    super(props);
    this.state = {
      username: props.username
    };
  }

  render() {
    const { username } = this.state;
    return (
      <TeamContext.Consumer>
        { teamContext => (
          <a
            className="c-user"
            href={`${teamContext.serverUrl}/direct/${username}`}
            rel="noopener noreferrer"
            target="_blank"
          >
            {`@${username}`}
          </a>
        )}
      </TeamContext.Consumer>
    );
  }
}
