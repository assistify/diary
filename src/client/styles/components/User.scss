.avatar > :first-child {
    border-radius: 50%
}