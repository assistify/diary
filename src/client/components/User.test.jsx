import React from 'react';
import renderer from 'react-test-renderer';
import User from './User';

it('renders correctly', () => {
  const tree = renderer
    .create(<User username="testuser" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
