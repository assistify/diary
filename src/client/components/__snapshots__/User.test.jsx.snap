// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="c-user"
  href="/direct/testuser"
  rel="noopener noreferrer"
  target="_blank"
>
  @testuser
</a>
`;
